include/wil/cppwinrt_helpers.h

```diff
@@ -0,0 +1,220 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+
+#ifndef __WIL_CPPWINRT_HELPERS_DEFINED
+#define __WIL_CPPWINRT_HELPERS_DEFINED
+
+#include <memory>
+
+/// @cond
+#if defined(_RESUMABLE_FUNCTIONS_SUPPORTED)
+#include <experimental/coroutine>
+namespace wil::details
+{
+    template<typename T = void> using coroutine_handle = std::experimental::coroutine_handle<T>;
+}
+#elif defined(__cpp_lib_coroutine) && (__cpp_lib_coroutine >= 201902L)
+#include <coroutine>
+namespace wil::details
+{
+    template<typename T = void> using coroutine_handle = std::coroutine_handle<T>;
+}
+#else
+namespace wil::details
+{
+    template<typename T = void> struct coroutine_handle

```

Comment:
Actually, I realized I shouldn't do this because it will create ODR problems if somebody links together some files with `/await` enabled, and other files without.